src/boot/store.js

```diff
@@ -41,14 +41,18 @@ export const storeKeys = ['migrations', 'accounts', 'drafts', 'outbox', 'setting
  */
 // prettier-ignore
 export const cacheKeys = [
-  'mute', 'narrows', 'realm', 'streams', 'subscriptions', 'unread', 'userGroups', 'users',
+  'messages', 'mute', 'narrows', 'realm', 'streams', 'subscriptions', 'unread', 'userGroups', 'users',
 ];
 
 const migrations = {
   '0': state => {
     AsyncStorage.removeItem('reduxPersist:messages');
     return state;
   },
+  '1': state => ({
```

Comment:
Hmm. What happens when a user upgrades across *both* of these migrations at once? Which of course will be the usual case for users that aren't us.

I think the answer is that they'll get a `messages` subtree which is the old `messages`. Which means we should drop the `messages` subtree in one of these migrations.

I think the logical place for that is migration 0, the one that removed `messages`. When we originally did that migration, we reasoned that because there was no `messages` top-level reducer anymore, redux-persist wouldn't even look at `messages` in the persisted store, so the state passed to the migrations would already lack `messages`; but that isn't true anymore.

(If that's right, then one lesson for future migration style is that we should have done that deletion in the original migration in the first place, even though it seemed redundant at the time -- that way in the event of re-using the key later, everything would work without depending on us noticing the re-use and going back and adding the deletion.)

So I guess something like
```
  const { messages, ...restState } = state;
  return restState;
```
is in order.

Reply:
Aah, yeah, good catch. 

>   const { messages, ...restState } = state;
  return restState;

I agree this is the right thing to do, in migration 0.